BlueprintUI/Sources/Layout/Flow.swift

```diff
@@ -0,0 +1,372 @@
+/// │ │   4   ││  5  ││     6     │       │
+/// │ └───────┘└─────┘└───────────┘       │
+/// └─────────────────────────────────────┘
+/// ```
+public struct Flow: Element {
+
+    /// How to horizontally align items in the row when there is extra space.
+    public var horizontalAlignment: HorizontalAlignment
+
+    /// How to vertically align items which there is extra space.
+    public var rowAlignment: RowAlignment
+
+    /// Space between items within a row.
+    public var horizontalSpacing: CGFloat
+
+    /// Space between rows.
+    public var rowSpacing: CGFloat
+
+    /// The child elements of the flow layout to be laid out.
+    public var children: [Child]
+
+    /// Creates a new flow layout with the provided parameters.
+    public init(
+        horizontalAlignment: HorizontalAlignment = .leading,
+        horizontalSpacing: CGFloat = 0,
+        rowAlignment: RowAlignment = .center,
+        rowSpacing: CGFloat = 0,
+        @ElementBuilder<Flow.Child> _ children: () -> [Flow.Child]
+    ) {
+        self.horizontalAlignment = horizontalAlignment
+        self.horizontalSpacing = horizontalSpacing
+        self.rowAlignment = rowAlignment
+        self.rowSpacing = rowSpacing
+        self.children = children()
+    }
+
+    // MARK: Element
+
+    public var content: ElementContent {
+        .init(
+            layout: Layout(
+                horizontalAlignment: horizontalAlignment,
+                rowAlignment: rowAlignment,
+                horizontalSpacing: horizontalSpacing,
+                rowSpacing: rowSpacing
+            )
+        ) {
+            for child in children {
+                $0.add(traits: child.traits, key: child.key, element: child.element)
+            }
+        }
+    }
+
+    public func backingViewDescription(with context: ViewDescriptionContext) -> ViewDescription? {
+        nil
+    }
+}
+
+extension Flow {
+
+    /// How to horizontally align items in the row when there is extra space.
+    public enum HorizontalAlignment: CaseIterable {
+
+        /// Items are aligned with the leading edge.
+        case leading
+
+        /// Items are centered within the remaining space.
+        case center
+
+        /// Items are aligned with the trailing edge.
+        case trailing
+    }
+
+    /// How to vertically align items which there is extra space.
+    public enum RowAlignment: CaseIterable {
+
+        /// Shorter items are stretched to fill the height of the tallest item.
+        case fill
+
+        /// Shorter items are aligned to the top of the row.
+        case top
+
+        /// Shorter items are vertically aligned within the row.
+        case center
+
+        /// Shoter items are aligned to the bottom of the row.
+        case bottom
+    }
+
+    /// A child placed within the flow layout.
+    public struct Child: ElementBuilderChild {
+
+        /// The key used to optionally unique the child item.
+        public var key: AnyHashable?
```

Comment:
Need to add some element extensions for setting this key, like we do for `stackLayoutChild`